### DISCUSSION TITLE
Galaxy Buds + Issue

### DISCUSSION THREAD

**Discussion by beforesq** [timschneeb/GalaxyBudsClient#69] [General]
Hello. I paired my Galaxy Buds + on the BudsClient, but I am not able to edit the sound settings in equalizer or turn on ambient sound . Did I missed something?

**Comment by ArthurWolfhound** [comment]
Your issue does not meet standard bug report template. If you believe that "Bug report" template does not fit for your question, please be kind to the developer and contributors here and at least add video capture or a series of screenshots to clarify your problem.

Sincerely @ArthurWolfhound

**Comment by beforesq** [comment]
> Your issue does not meet standard bug report template. If you believe that "Bug report" template does not fit for your question, please be kind to the developer and contributors here and at least add video capture or a series of screenshots to clarify your problem.
> 
> Sincerely @ArthurWolfhound

You don't need a screenshot to understand my problem. I just can't use the earbuds features while they're connected to the pc.

**Comment by timschneeb** [comment]
What happens when you change any setting? Does it just do nothing or do the sound settings reset immediately?

Also, which firmware version are you running on your Buds+? You don't need to type out the whole version string, just the last four characters are enough. (example: `R175XXU0ATH7` => `ATH7`).
In the GalaxyBudsClient, you can find this under "`System Information`" and on Android, you can find this data under "`About earbuds > Software information`"

**Comment by beforesq** [comment]
> What happens when you change any setting? Does it just do nothing or do the sound settings reset immediately?
> 
> Also, which firmware version are you running on your Buds+? You don't need to type out the whole version string, just the last four characters are enough. (example: `R175XXU0ATH7` => `ATH7`).
> In the GalaxyBudsClient, you can find this under "`System Information`" and on Android, you can find this data under "`About earbuds > Software information`"


Nothing happens when I change the settings in the app. I am using a Bluetooth adapter to pair my earbuds , but they're pairing instantly. 
My earbuds version is :  ATK1

**Comment by beforesq** [comment]
@ThePBone I don't understand why do I have 2 different earbuds connected in windows: 
![7tB1bit](https://user-images.githubusercontent.com/60829541/101449200-04b3b900-3931-11eb-87c3-c200850673b8.png)

Edit: if I switch to the stereo earbuds, the quality of the sound is really good, the same as on my phone, but if I switch to "Hands-Free AG", the sound is really bad, it has not bass. ( Sounds like 15$ buds )

**Comment by ArthurWolfhound** [comment]
> @ThePBone I don't understand why do I have 2 different earbuds connected in windows:
> 
> Edit: if I switch to the stereo earbuds, the quality of the sound is really good, the same as on my phone, but if I switch to "Hands-Free AG", the sound is really bad, it has not bass. ( Sounds like 15$ buds )

If I am correct (though I can be not, but I have Headphones and Headset too) that should be a standard behavior for all wireless earbuds. Without going into a lot of technical details, a lot of which I myself don't understand entirely - Headphones is a virtual profile for music streaming, Headset is a virtual profile for voice calls such as Zoom, Skype, Discord and others.

**Comment by beforesq** [comment]
> > @ThePBone I don't understand why do I have 2 different earbuds connected in windows:
> > Edit: if I switch to the stereo earbuds, the quality of the sound is really good, the same as on my phone, but if I switch to "Hands-Free AG", the sound is really bad, it has not bass. ( Sounds like 15$ buds )
> 
> If I am correct (though I can be not, but I have Headphones and Headset too) that should be a standard behavior for all wireless earbuds. Without going into a lot of technical details, a lot of which I myself don't understand entirely - Headphones is a virtual profile for music streaming, Headset is a virtual profile for voice calls such as Zoom, Skype, Discord and others.

Headphones doesnt work anymore. I can't hear anything if I switch to Headphones. Only Headset works at the moment. But the quality on headset is horrible, as I said, sounds like 15$ earbuds.

**Comment by timschneeb** [comment]
Could you delete the earbuds in Windows and pair them again?

That sounds like there are issues with your Bluetooth driver/adapter. Updating/reinstalling these drivers might help as well.

**Comment by beforesq** [comment]
> Could you delete the earbuds in Windows and pair them again?
> 
> That sounds like there are issues with your Bluetooth driver/adapter. Updating/reinstalling these drivers might help as well.

I had a cd in my bluetooth adapter box, but I dont have a cd rom. I'll try to find the driver and i will edit this comment. Thank you. The problem with the app is the same. Nothing happens when  I change the settings.
Edit 1 : I installed the driver from asus.com. Testing the buds right now.
Edit 2 : There's no difference. The sound quality is extremely bad, no bass, no sound effects.
Edit 3: This is my bluetooth adapter: USB Bluetooth ASUS USB-BT400

**Comment by beforesq** [comment]
@ThePBone Something weird happened today. I was in a meet, with my big on ear headphones connected to the pc. I connected my buds to go in the kitchen, and I noticed that the ambient sound , the gestures and the equalizer were working on the pc. I don't know how I did that, because I couldn't do it the 2nd time. I had the bluetooth turned on , on the pc and on the phone.Your app was off on pc.
Any idea?

**Comment by ArthurWolfhound** [comment]
I am sorry, this thread feels like a train wreck. Would you mind starting off scratch completely? I am providing instructions for complete Buds reinstall, alongside with system clean-up process. You **do not** need to reinstall system fully, just Buds, Bluetooth drivers, and use some system commands to perform a system integrity check.

# 1. Delete your Galaxy Buds and Bluetooth from your System
1. Use PC Settings to remove your Buds, both the Headphones and Headset
2. Check if you have any Buds alike devices in other categories, Windows can sometimes have extremely strange bugs that make zero sense to anyone.
3. Press Win+X
4. Find Device Manager in the pop-up menu and open it
5. View -> Show hidden devices
6. Check following categories and if any devices that have Galaxy **Buds** or your Bluetooth adapter in the name exist, remove them one by one with right-click, uninstall device:
6.1. Audio inputs and outputs
6.2. Bluetooth
6.3. Network adapters
6.4. Sound, video and game controllers
7. Reboot.

# 2. Perform a System Integrity Check
1. Open elevated to Admin Windows Powershell. Depending from system to system you may be able to press Win+X and find `Windows Powershell (Admin)` there. If not - use Windows search, right-click, open as Admin.
2. Run command `DISM.exe /Online /Cleanup-image /Restorehealth`, do mind that you need internet connection to use this command. At this stage operating system will analyze if any files are missing or corrupted and then will download them from Microsoft Servers.
3. After successful completion (for which I pray), you will need to run command `sfc /scannow`, this time it is going to apply files to your system.
4. Reboot.

# 3. Install your devices back
As we are being cautious, do reboot after every device install.

1. Install your Bluetooth adapter
2. Install your Buds and try to run them without application present. To check how your Buds work you may try to enable "Tap on Buds body to raise or lower volume" feature in Buds Labs in Galaxy Wearable on mobile, and then try to use this feature on PC. Also I should note, that if you have Cortana available on your system, and one of your Buds is configured to call Voice Assistant - this should work. At least it works on my system without any questions raised.
3. Try to run application now and check if you are able to do anything. 


# Finale
If any of the system integrity checks will fail or you will get any errors on any stage, you can post screenshots of errors here, but we do not guarantee that any of us will be able to help, and due to the fact that such system integrity repair commands are extensive enough to fix problems - there might be a possibility of a serious corruption that may be caused as by any other third-party software, as by famous quality of Microsoft Updates for Windows. So to conclude - if you get any errors, there might be a possibility that you **will need** to do a **clean install** of your system.

**Comment by beforesq** [comment]
@ArthurWolfhound [==========================100.0%==========================] The restore operation completed successfully.
The operation completed successfully.
PS C:\WINDOWS\system32> sfc /scannow

Beginning system scan.  This process will take some time.

Beginning verification phase of system scan.
Verification 100% complete.

Windows Resource Protection did not find any integrity violations.
PS C:\WINDOWS\system32>

Rebooting and trying my earbuds in a moment.


Edit: The app works really well now, after reintalling the buds and the adapter, but, when I open discord, and switch to the buds, I can't hear anything in them. Do you know why? I can hear perfectly in the buds, I got ambient sound and everything. Thanks.

**Comment by ArthurWolfhound** [comment]
Glad to solve the problem! Discord...is being a jerk again, I am constantly having the same issue with it.

1. Check if you by any wild chance have Push-to-Talk on. Got me quite a lot of times.
2. Check Discord settings for both Input and Output to be set for Headset.

**Comment by ArthurWolfhound** [comment]
Oh wait, I might have misunderstood you. Have you meant that when you switch to Discord you lose Ambient Sound effect?

**Comment by beforesq** [comment]
@ArthurWolfhound When I change the input to Galaxy Buds + , the earbuds are stopping. I mean, the sound is stopping, and the gestures, everything.

**Comment by beforesq** [comment]
![download](https://user-images.githubusercontent.com/60829541/101635238-598b2880-3a32-11eb-9636-14bf4c872d2b.png)
![RsBFQgF](https://user-images.githubusercontent.com/60829541/101635243-5a23bf00-3a32-11eb-936e-abd279573e89.png)


In control panel, I got 2 "Galaxy Buds" devices in the playback tab:
![GUvIbUP](https://user-images.githubusercontent.com/60829541/101635372-8d664e00-3a32-11eb-9077-249b862af219.png)

When I press on the Recording tab, just pressing on it, not changing the default device, the earbuds are stopping again.
![Jz4UMRz](https://user-images.githubusercontent.com/60829541/101635566-ce5e6280-3a32-11eb-8df0-6aee54fea595.png)

**Comment by ArthurWolfhound** [comment]
Yeah I've got it. That's actually not solvable. If you recall one of my messages earlier where I talked about two virtual devices - Discord takes monopolist control after Headset because Discord is designed to work with Buds "Call protocol". And while Discord is using your Buds, they stop responding - I've just tested it myself. You actually could have experienced that yourself when you've got Buds with Ambient on and somebody called you - ambient turns off the moment you take the call.

And if I am correct there's no way to turn Discord to Headphones instead of Headset because Discord doesn't understand how to work with Headphones virtual protocols.

@ThePBone am I correct in all that tech stuff? Btw, we can say the main issue is solved.

Oooh, and better not mess with that sound devices window, it is so weird. I can't really explain, just better don't touch things there.

**EDIT:** But the sound of yourself should work at least. Try Discord settings, set both input/output to Headset, press Let's Check button and start speaking - you should hear yourself. If not then choose Headset in volume tray menu.

**Comment by beforesq** [comment]
Yeah, I can use my earbuds just to hear people on discord, but I can't use the bud's microphones. That's okay because I'll use the microphone from my on-ear headphones. Thank you very much. If you find a fix for this in the future, reply to this thread. And thank you for providing me the instructions that I needed.

**Comment by ArthurWolfhound** [comment]
That should not work like that. I've just tested my Buds and when I was 1.5m away from PC Discord used my Buds Mic to playback my voice.

When Headphones are chosen in the Volume Manager
![image](https://user-images.githubusercontent.com/16789106/101636765-74f73300-3a34-11eb-9efc-2c7ea82e7042.png)


You should have this settings in Sound menu
![image](https://user-images.githubusercontent.com/16789106/101636800-7e809b00-3a34-11eb-8139-359a8427caf8.png)
![image](https://user-images.githubusercontent.com/16789106/101636816-82acb880-3a34-11eb-9854-a6894029fabe.png)


And both input/output set to Headset in Discord - that way it **should** work. If not - then I am out of ideas...

**Comment by beforesq** [comment]
If I set the input to galaxy buds, they're stopping. I got the same settings in control panel like you. It's okay anyway, I can use my microphone from the big headphones, but, you know, I have to stay near it to capture my voice.

**Comment by beforesq** [comment]
One more question, @ArthurWolfhound . Is this feature safe to use? I don't have it in the phone app:
![dvxB6pX](https://user-images.githubusercontent.com/60829541/101637923-0d41e780-3a36-11eb-93c2-a8f4c96873c1.png)

**Comment by ArthurWolfhound** [comment]
This one is actually in the Labs part of Galaxy Wearable in mobile

![Screenshot_20201209-163056_Galaxy Buds+.jpg](https://user-images.githubusercontent.com/16789106/101642663-ef778100-3a3b-11eb-9737-6b3e98917929.jpg)

I've never used it. And have no idea what it does. But in Windows Client it exists only to be able to switch it on and off. It applies only to Samsung devices.

**Comment by timschneeb** [comment]
@beforesq Gaming Mode is safe to use, however, it will only have an effect on Samsung mobile phones that support the `GameOptimizingService`.

> am I correct in all that tech stuff? 

@ArthurWolfhound Yeah, that's pretty accurate.
By the way, after all your contributions, I think it wouldn't be a bad idea to give you write access for this repo. I've sent you a collaborator invite, so you can accept it if you want.

**Comment by beforesq** [comment]
Oh, Okay. I don't have a Samsung Phone at the moment. Thanks.

@ArthurWolfhound Do you have small problems with the latency, or it's my bluetooth adapter? In games, I got a 1 second sound delay. Is it because the adapter has bluetooth 4.0 and the earbuds got 5.0?

**Comment by ArthurWolfhound** [comment]
@beforesq, I have a laptop so my adapter is on motherboard directly, so I would say yes - USB can make a latency delay. About Bluetooth version - it should not be creating issues if you're right nearby your PC, because Bluetooth has backwards compatibility. That may be caused if you're plugging adapter in the back of your PC, electromagnetic fields, too much electric noise nearby, maybe you're in the radio crowded place, like say everybody has not just their WiFi router turned on, but two phones and tablets constantly. Also there's an actual possibility that if you have **yours** WiFi router right near PC it can actually cause issues, though, I had those on Samsung U Flex.

TL;DR: There's like infinite possible explanations, but Bluetooth version should not be a problem.

**Comment by beforesq** [comment]
@ArthurWolfhound I plugged the adapter in the front panel, in the usb 3.0 port and my wifi router is far from my room. I remember that I fixed the latency problem from control panel. I found this fix on reddit. But If I disable that thing, I wont be able to talk with the earbuds on the pc, so , it's not a good fix.
